### ISSUE TITLE
project-velero/kopia: ship binaries or containers for debugging

### ISSUE THREAD

**Issue by kaovilai** [vmware-tanzu/velero#8410]
**Describe the problem/challenge you have**

Velero uses [a fork that is not 100% bug for bug with kopia/kopia](https://github.com/kopia/kopia/compare/master...project-velero:kopia:v0.17.0-velero-patch) and it would be great if we could produce artifacts similarly to  those found at https://github.com/kopia/kopia/releases/tag/v0.17.0 in project-velero/kopia fork.

In addition, if these binaries are added to node-agent containers (or sidecar) where it will have access to the same ENV, creds, it would be great for debugging kopia repositories without additional credential steps.

**Describe the solution you'd like**

Inherit/enable the same github actions of upstream and tag for every velero release that uses new fork kopia tag

**Anything else you would like to add:**

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by Lyndon-Li** [comment]
This is a rational requirement.
But we don't need to do this immediately necessarily. The current code in project-velero/kopia is still compatible with the upstream Kopia client, as long as the right version is being used.
We have clarified the base Kopia version being used by each Velero release in the release notes. So users could find the right version there and use the upstream client of that version to connect to Velero repository.

In future, along with the changes of project-velero/kopia, when we see a particular version of client is required, we will deliver it somehow.

**Comment by weshayutin** [comment]
I also agree this is probably a must have.  I do wonder though if customers will get themselves into trouble with it.  I wonder if we could detect whether or not a customer connected to a repo and ran a maintenance job vs. the velero maint job.  For example the case where the customer actually caused a problem vs. resolved something, is there a way for us to cover ourselves and know what happened?

**Comment by kaovilai** [comment]
> if we could detect whether or not a customer connected to a repo and ran a maintenance job vs. the velero maint job.

sounds like kopia enhancement to log access if it isn't already implemented.

**Comment by Lyndon-Li** [comment]
> I do wonder though if customers will get themselves into trouble with it

Once users connects Velero repo through Kopia client, they could do many more critical operations So even if we could detect the self-run maintenance, it is hard to say we can detect all others, and it may not be worthy doing that.
